colossalai/kernel/base_kernel_loader.py

```diff
@@ -0,0 +1,48 @@
+import platform
+from abc import ABC, abstractmethod
+from typing import Dict, List
+
+import torch
+
+from .extensions.base_extension import BaseExtension
+
+
+class BaseKernelLoader(ABC):
+    """
+    Usage:
+        kernel_loader = KernelLoader()
+        kernel = kernel_loader.load()
+    """
+
+    def __init__(self, extension_map: Dict[str, BaseExtension], supported_device: List[str]):
+        self._extension_map = extension_map
+        self._supported_device = supported_device
+
+    def run_checks(self):
+        # run supported device check and other possible checks
+        pass
+
+    @abstractmethod
+    def fetch_kernel(self):
+        pass
+
+    def load(self):
+        self.run_checks()
+        return self.fetch_kernel()
+
+    def _is_x86_available(self) -> bool:
+        return platform.processor() == "x86_64"
```

Comment:
Should be careful when using `platform.processor` as it may return unexpected values.

<img width="822" alt="Screenshot 2023-12-11 at 15 36 49" src="https://github.com/hpcaitech/ColossalAI/assets/31818963/069d1d61-4c68-4e5b-8837-913eea49fc34">

As it can return the same values as `platform.machine()`, there will be more than one value that represents `arm` or `x86` based on this [post](https://stackoverflow.com/questions/45125516/possible-values-for-uname-m).

Reply:
ok, i will add other possible values for them